docs/data_sharing_guide.md

```diff
@@ -0,0 +1,338 @@
+- `unshare_table(namespace, table_name, from_users, from_groups)` - Unshare SQL tables
+
+**Pre-Initialized Client:**
+- `governance` - Pre-initialized `DataGovernanceClient()` instance for advanced operations
+
+**Other Auto-Imported Functions:**
+- `get_spark_session()` - Create Spark sessions with Delta Lake support
+- Plus many other utility functions for data operations
+
+### Quick Start
+
+```python
+# Check your workspace information
+workspace = get_my_workspace()
+print(f"👤 Username: {workspace.username}")
+print(f"🏠 Home paths: {workspace.home_paths}")
+print(f"👥 Groups: {workspace.groups}")
+print(f"📁 Accessible paths: {len(workspace.accessible_paths)}")
+
+# Check service health
+health = check_governance_health()
+print(f"🔍 Service status: {health.status}")
+
+# Get your credentials (automatically sets environment variables)
+credentials = get_minio_credentials()
+print(f"🔐 Credentials loaded for: {credentials.username}")
+```
+
+## Core Information Functions
+
+### Service Health and Credentials
+
+```python
+# Check governance service status
+health = check_governance_health()
+print(f"Service status: {health.status}")
+
+# Get MinIO credentials (automatically sets MINIO_ACCESS_KEY and MINIO_SECRET_KEY)
+credentials = get_minio_credentials()
+print(f"Username: {credentials.username}")
+print(f"Access key: {credentials.access_key}")
+print(f"Secret key: {credentials.secret_key}")
+
+# Get your SQL warehouse prefix
+sql_warehouse = get_my_sql_warehouse()
+print(f"SQL warehouse prefix: {sql_warehouse.sql_warehouse_prefix}")
+
+# The pre-initialized governance client is also available
+print(f"Governance client ready: {governance is not None}")
+```
+
+### Workspace and Policy Information
+
+```python
+# Get comprehensive workspace information
+workspace = get_my_workspace()
+print(f"Username: {workspace.username}")
+print(f"Home paths: {workspace.home_paths}")
+print(f"Groups: {workspace.groups}")
+print(f"Total accessible paths: {len(workspace.accessible_paths)}")
+
+# Get detailed policy information
+policies = get_my_policies()
+print(f"Home policy: {policies.user_home_policy.policy_name}")
+print(f"System policy: {policies.user_system_policy.policy_name}")
+print(f"Group policies: {len(policies.group_policies)}")
+```
+
+## Sharing SQL Warehouse Tables
+
+### Share Tables with Users
+
+```python
+# Share a table with specific users
+response = share_table(
+    namespace="research",
+    table_name="climate_data",
+    with_users=["bob", "alice"]  # kbase usernames
+)
+
+print(f"Shared with users: {response.shared_with_users}")
+print(f"Success count: {response.success_count}")
+if response.errors:
+    print(f"Errors: {response.errors}")
+```
+
+### Share Tables with Groups
+
+```python
+# Share a table with groups
+response = share_table(
+    namespace="analytics", 
+    table_name="user_metrics",
+    with_groups=["data-science-team", "research-group"]
+)
+
+print(f"Shared with groups: {response.shared_with_groups}")
```

Comment:
For this and the next examples should response.errors be checked?

Reply:
👍